bin/cylc

```diff
@@ -107,7 +102,8 @@ __HELP__
     fi
     # Deal with cases like 'cylc --help [COMMAND/CATEGORY]'
     if [[ $# -gt 1 && -f "$(ls "cylc-$2"* 2>'/dev/null')" ]]; then
-        local COMMAND="${CYLC_HOME_BIN}/$(ls "cylc-$2"*)"
+        COMMAND="${CYLC_HOME_BIN}/$(ls "cylc-$2"*)"
+        local COMMAND
```

Comment:
Can't see why this is better.

Reply:
Originally, if the `ls` command fails, the script won't itself fail because the `local COMMAND` doesn't fail. To catch errors from each command, you need to separate them into multiple lines.

Reply:
I understand why we want to do this for `export`. But for `local` as well?

And does the first assignment command turn `COMMAND` into a module variable briefly before the second command turn it back to a local one?

Reply:
Good question. I don't know. Probably? I tend to put the `local` or `export` above variable assignment.

Reply:
> Can't see why this is better.

So this is about masking the return value, tldr when using subshells do the assignment and declaration on different lines.

I put the declaration after the assignment for consistency with the `export` usage but admittedly that seems a bit mad, I'll swap the order around.

https://github.com/koalaman/shellcheck/wiki/SC2155